src/psyclone/psyGen.py

```diff
@@ -589,39 +598,43 @@ def first_access(self, arg_name):
         raise GenerationError("Failed to find any kernel argument with name "
                               "'{0}'".format(arg_name))
 
-    def unique_declns_by_intent(self, datatype):
+    def unique_declns_by_intent(self, argument_type):
         '''
         Returns a dictionary listing all required declarations for each
         type of intent ('inout', 'out' and 'in').
 
-        :param string datatype: the type of the kernel argument for the \
-                                particular API for which the intent is \
-                                required
+        :param string argument_type: the type of the kernel argument for the \
+                                     particular API for which the intent is \
+                                     required.
+
         :returns: dictionary containing 'intent' keys holding the kernel \
-                  argument intent and declarations of all kernel arguments \
-                  for each type of intent
-        :rtype: dict
+                  arguments as values for each type of intent.
+        :rtype: dict of :py:class:`psyclone.psyGen.KernelArgument`
+
         :raises GenerationError: if the kernel argument is not a valid \
-                                 datatype for the particular API.
+                                 argument type for the particular API.
 
         '''
-        if datatype not in VALID_ARG_TYPE_NAMES:
+        if argument_type not in VALID_ARG_TYPE_NAMES:
             raise GenerationError(
```

Comment:
Again, please could you make this an `InternalError`.

Reply:
Made and updated docstring.